### PR TITLE
Add CI definition for VSTS for macOS

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Add CI definition for VSTS for macOS
  - Add VSTS YAML for mac
  - Add a tag to skip tests which are not working in VSTS (I'm working on a fix for this issue.)
  - Add a function to upload log files during a test on VSTS

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by TravisEz13** [comment]
There is a bug in VSTS when you first add the definition.  The run from my branch is here: https://powershell.visualstudio.com/PowerShell/_build%2Fresults?buildId=258&_a=summary

**Comment by TravisEz13** [comment]
The new job for the update code is here:  https://powershell.visualstudio.com/PowerShell/_build/results?buildId=273&_a=summary&view=logs

**Review by dantraMSFT — APPROVED** [review]
I can sign off on the macos tests and the PSSyslog.psm1 changes.

File: vsts-macos-ci.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    displayName: Set Build Name for Non-PR
+    condition: ne(variables['Build.Reason'], 'PullRequest')
+
+# Several scripts require TMPDIR to exist
+# Set it to AGENT_TEMPDIRECTORY so that we know it exists
+  - powershell: |
+      Write-Host "##vso[task.setvariable variable=TMPDIR]$env:AGENT_TEMPDIRECTORY"
+    displayName: Set TMPDIR to AGENT_TEMPDIRECTORY
+    condition: succeededOrFailed()
+
```
Comment (daxian-dbw) [inline vsts-macos-ci.yml]: Trailing spaces at line 27 (reported by CodeFactor)

File: vsts-macos-ci.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+name: PR-$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:.rr)
```
Comment (daxian-dbw) [inline vsts-macos-ci.yml]: It's better to move this file in a folder, like `vsts-ci`, so we don't have more and more files under the repository root.

**Review by daxian-dbw — APPROVED** [review]
LGTM, with a few minor comments.

File: vsts-macos-ci.yml (hunk 2 above)
Comment (TravisEz13) [reply vsts-macos-ci.yml]: moved to a folder `.vsts-ci`

File: vsts-macos-ci.yml (hunk 1 above)
Comment (TravisEz13) [reply vsts-macos-ci.yml]: removed

File: vsts-macos-ci.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      }
+    displayName: Publish Artifacts
+    condition: succeeded()
+
+  - powershell: |
+      Get-ChildItem -Path Test*.xml -Recurse | Select-Object -ExpandProperty FullName | ForEach-Object {
+        Write-Host "##vso[artifact.upload containerfolder=testResults;artifactname=testResults]$_"
+      }
+    displayName: Publish Test Results
+    condition: succeeded()
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply vsts-macos-ci.yml]: changed

File: vsts-macos-ci.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  - task: PublishTestResults@2
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply vsts-macos-ci.yml]: Added comments

**Comment by TravisEz13** [comment]
link to latest run https://powershell.visualstudio.com/PowerShell/_build/results?buildId=285&_a=summary&view=logs

**Comment by TravisEz13** [comment]
latest VSTS run
https://powershell.visualstudio.com/PowerShell/_build/results?buildId=294&_a=summary&view=logs

**Pull request merged**